template/java8/model/src/main/java/com/openfaas/model/Request.java

```diff
@@ -52,6 +58,47 @@ public String getQueryRaw() {
 	public Map<String, String> getQuery() {
     	return queryParameters;
     }
+
+    @Override
+    public String getPath() {
+        return path;
+    }
+
+    @Override
+    public Map<String, String> getPathParameters() {
+        return pathParameters;
+    }
+
+    private Map<String, String> parsePathParameters() {
+        Map<String, String> res = new HashMap<String, String>();
+        if (path != null) {
+            String[] tokens = path.split("/");
+
+            String[] params = new String[0];
+            if (tokens.length > 1 && (tokens[0].equals("function") || tokens[0].equals("async-function"))) {
```

Comment:
Why do we reference `function` / `async-function`? This shouldn't be getting passed in.